core/common/src/main/java/alluxio/worker/block/BlockWorker.java

```diff
@@ -312,6 +313,14 @@ void requestSpace(long sessionId, long blockId, long additionalBytes)
    */
   void asyncCache(AsyncCacheRequest request);
 
+  /**
+   * Submits the async cache request to async cache manager to execute.
+   *
+   * @param request the async cache request
+   * @return cache result, succeed or not
+   */
+  boolean cache(CacheRequest request);
```

Comment:
I don't understand the `boolean` return value. It's already a synchronized operation. I think we should just throw `IOException` when it fails with something. It should be on client to handle it.

Reply:
I like the idea of just throw IOException, avoid lots of duplications caused by boolean return value.